src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -116,6 +140,31 @@ private UUID saveNewLetter(ILetterRequest letter, String messageId, String servi
         return id;
     }
 
+    private byte[] encryptZipContents(
+        ILetterRequest letter,
+        String serviceName,
+        UUID id,
+        byte[] zipContent,
+        LocalDateTime createdAt
+    ) {
+        Asserts.notNull(encryptionPublicKey, "encryptionPublicKey");
+
+        String zipFileName = ZipFileNameHelper.generateName(letter.getType(), serviceName, createdAt, id);
+
+        try {
+            PGPPublicKey pgpPublicKey = PgpEncryptionUtil.loadPublicKey(encryptionPublicKey.getBytes());
```

Comment:
instead of doing it for every letter, could this be once? perhaps in service constructor?